### PR TITLE
Add k8s-ruby/k8s-ruby as a community-maintained client libraries

### LINKED ISSUE #27458: Client Libraries
They `ruby-k8sclient` library is mildly defunct now 

The following project: https://github.com/k8s-ruby/k8s-ruby/ has been forked and is meant to replace the former per https://github.com/kontena/k8s-client/issues/172

Please update the documentation

### PR DESCRIPTION
**Pull request by shuuji3**
Resolves #27458.

Page preview: https://deploy-preview-27502--kubernetes-io-master-staging.netlify.app/docs/reference/using-api/client-libraries/

How to test:
- Check if k8s-ruby/k8s-ruby exists in the table of client libraries.

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
/approve
LGTM, but I think the sort order could use a tweak.

File: content/en/docs/reference/using-api/client-libraries.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ their authors, not the Kubernetes team.
 | Ruby                 | [github.com/abonas/kubeclient](https://github.com/abonas/kubeclient) |
 | Ruby                 | [github.com/Ch00k/kuber](https://github.com/Ch00k/kuber) |
 | Ruby                 | [github.com/kontena/k8s-client](https://github.com/kontena/k8s-client) |
+| Ruby                 | [github.com/k8s-ruby/k8s-ruby](https://github.com/k8s-ruby/k8s-ruby) |
```
[the remark this answers is not in the document]
Comment (shuuji3) [reply content/en/docs/reference/using-api/client-libraries.md]: That's right. Thanks for the suggestion.

**Pull request merged**